### PR TITLE
Conventions for our open source gradle plugins

### PR DESCRIPTION
**Pull request by stormbeta**

### CODE REVIEW

File: src/main/groovy/com/readytalk/gradle/plugins/CiLifecyclePlugin.groovy [hunk 1]
```diff
@@ -99,10 +99,66 @@ class CiLifecyclePlugin implements Plugin<Project>, PluginUtils {
...
 
+      withPluginId('com.github.kt3k.coveralls') {
+        tasks.ci.dependsOn tasks.coveralls
+        tasks.coveralls { cover ->
+          onlyIf { infoExt.isCI.toBoolean() }
+        }
+      }
+
+      //Open source plugin project conventions
+      withPlugins(['nu.studer.plugindev', 'com.jfrog.artifactory', 'com.jfrog.bintray']) {
```
Comment (stormbeta) [inline src/main/groovy/com/readytalk/gradle/plugins/CiLifecyclePlugin.groovy:110]: Only gradle plugin projects have a reason to apply the plugindev plugin, making this a reasonable set of triggers for enabling gradle plugin project conventions for publishing.

**Comment by sgoings** [comment]
I'm wondering if maybe it would be smarter to have a highly opinionated plugin for external projects (plugins, etc.) vs. trying to predict what people want. I really like the model that Gradle chose to take with their base -> java-base -> java style opinion stacking. It seems like the comment on line 70:

`//TODO: Include more conventions and split into separate plugin class`

should be the route we take instead of trying to auto-determine what project the ci project is being applied to. Thoughts?

The plugin-dev plugin totally makes sense as a trigger for gradle plugins, but there are projects I'm using that aren't Gradle plugins that make sense to set up the artifactory/oss.jfrog connection (see: Avian).

**Comment by dougborg** [comment]
I agree. I think it may be best to provide a set of plugins which can be explicitly applied, rather than get too smart about auto-applying them. They may be configured a bit differently based on what other plugins are applied, but let's not try to have one plugin for everything.

For example for an external oss project:

``` groovy
plugins {
 id 'java'
 id 'com.readytalk.ci-oss' version '0.2.0'
}
```

An internal java project may look something like this:

``` groovy
plugins {
 id 'java'
 id 'com.readytalk.ci-internal' version '0.2.0'
}
```

Under the hood, they both apply a common `readytalk-ci` plugin, but then add the appropriate opinions / conventions. Is this more along the lines of what you are talking about @sgoings?

**Comment by stormbeta** [comment]
Closing for now.

I agree we need a more extensible and clear way to define conventions, but I'd like to wait and see how the new model DSL in Gradle pans out before committing to a particular way of doing things. Moreover, I'd like to work on better integration with the nebula plugins first.

**Pull request closed without merging**